src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -165,6 +167,30 @@ public static void copyAll(@Nonnull Run<?,?> from, @Nonnull Run<?,?> to) throws
         FileUtils.copyDirectory(fromStorage, storage(to));
     }
 
+    /**
+     * Get the filenames included in a particular stash.
+     * @param build The build to look in.
+     * @param stashName The stash name to look up.
+     * @return A possibly empty list of filenames included in the given stash name in the given build. Unix separators will be used.
+     * @throws IOException If unable to interact with the stash files.
+     */
+    @Nonnull
+    public static List<String> stashFileList(@Nonnull Run<?,?> build, @Nonnull String stashName) throws IOException {
+        List<String> contents = new ArrayList<>();
```

Comment:
Why not `Set<String>`?

Reply:
`Set` is probably more technically correct, but I think `List` will be easier to work with due to positional access.